### PR TITLE
Changing the system role may undo adjustments you may have done.

### PR DESCRIPTION
**Pull request by mvidner**
A missing piece to complete https://trello.com/c/0mOtUy45/467-8-sle-12-sp2-p4-m-feature-317481-add-system-role-capabilities-to-installer , continuing #340.

### CODE REVIEW

File: test/select_system_role_test.rb [hunk 1]
```diff
@@ -64,6 +64,60 @@
 
         expect(subject.run).to eq(:back)
       end
+
+      context "when re-selecting a role" do
+        it "just proceeds on the same role" do
```
Comment (kobliha) [inline test/select_system_role_test.rb]: I'm missing another level of context: "no role has been selected before"

File: test/select_system_role_test.rb [hunk 2]
```diff
@@ -64,6 +64,60 @@
...
+
+          expect(Yast::Popup).to_not receive(:ContinueCancel)
+
+          expect(Yast::ProductFeatures).to receive(:ClearOverlay)
+          expect(Yast::ProductFeatures).to receive(:SetOverlay)
+
+          expect(subject.run).to eq(:next)
+        end
+
+        it "displays popup, and proceeds on Continue" do
```
Comment (kobliha) [inline test/select_system_role_test.rb]: Some role has been selected before

File: test/select_system_role_test.rb [hunk 3]
```diff
@@ -64,6 +64,60 @@
...
+          expect(Yast::Popup).to receive(:ContinueCancel)
+            .and_return(true)
+
+          expect(Yast::ProductFeatures).to receive(:ClearOverlay)
+          expect(Yast::ProductFeatures).to receive(:SetOverlay)
+
+          expect(subject.run).to eq(:next)
+        end
+
+        it "displays popup, and stays on Cancel" do
```
Comment (kobliha) [inline test/select_system_role_test.rb]: And what is different here? it should be documented in context

File: test/select_system_role_test.rb (hunk 1 above)
Comment (mvidner) [inline test/select_system_role_test.rb]: In all the tests in this file I am omitting a `context` that would contain only one `it` example.
But yes, I should say it "just proceeds if the re-selected role was the same as before"

**Comment by mvidner** [comment]
Test structure updated, thanks.

File: src/lib/installation/select_system_role.rb [hunk 4]
```diff
@@ -62,12 +68,22 @@ def dialog_content
...
     end
 
     def next_handler
       role_id = Yast::UI.QueryWidget(Id(:roles), :CurrentButton)
+
+      orig_role_id = self.class.original_role_id
+      if !orig_role_id.nil? && orig_role_id != role_id
+        # A Continue-Cancel popup
+        msg = _("Changing the system role may undo adjustments you may have done.")
+        Yast::Popup.ContinueCancel(msg) || return
```
Comment (lslezak) [inline src/lib/installation/select_system_role.rb]: Um, using `||` for control flow looks a bit weird to me, I'd prefer

```ruby
return unless Yast::Popup.ContinueCancel(msg)
```

**Comment by lslezak** [comment]
LGTM

**Pull request merged**